### ISSUE TITLE
pornHD extractor not working properly

### ISSUE THREAD

**Issue by operabeats** [rg3/youtube-dl#4893]
Hi 

I was trying download video from pornHD (http://www.pornhd.com/videos/2602/horny-blondes-really-into-facesitting-hd-porn-movie) and I get the following error:

>  File "/home/vaio/temp/youtube_dl/extractor/pornhd.py", line 56, in _real_extract
>    r"(?s)'sources'\s*:\s*(\{.+?\})\s*\}\);", webpage, 'sources')))
>  File "/home/vaio/temp/youtube_dl/extractor/common.py", line 524, in _search_regex
>    raise RegexNotFoundError('Unable to extract %s' % _name)
>RegexNotFoundError: Unable to extract sources; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

I tinkered with pornhd.py extractor file and found there was an error in regex for "source"
```
sources = json.loads(js_to_json(self._search_regex(
            r"(?s)'sources'\s*:\s*(\{.+?\})\s*\}\);", webpage, 'sources')))
```
and changed it to
```
        sources = json.loads(js_to_json(self._search_regex(
            r"(?s)'sources'\s*:\s*(\{.+?\})\s*\}", webpage, 'sources')))
```
then there was error with next block, so changed:
```
formats = []
        for container, s in sources.items():
            for qname, video_url in s.items():
                formats.append({
                    'url': video_url,
                    'container': container,
                    'format_id': '%s-%s' % (container, qname),
                    'quality': quality(qname),
                })
        self._sort_formats(formats)
```
to
```
        formats = []
        for qname, video_url in sources.items():
                formats.append({
                    'url': video_url,
                    'quality': quality(qname),
                })
        self._sort_formats(formats)
```
Now I am getting the following error:
>  File "/usr/lib/python2.7/cookielib.py", line 604, in eff_request_host
    erhn = req_host = request_host(request)
  File "/usr/lib/python2.7/cookielib.py", line 592, in request_host
    host = request.get_header("Host", "")
AttributeError: '_PseudoRequest' object has no attribute 'get_header'

Please help me getting this resolved.
(I am new to python / js / json and first time posting on github, please pardon me if there are any mistakes. Let me know if any further info is required)

(if this helps someone, i used http://pythex.org/ to understand the regex!)

**Comment by robin007bond** [comment]
Can confirm this. I tried to figure it out too, but haven't been able to get far.

**Comment by phihag** [comment]
Indeed, this has already been fixed in youtube-dl 2015.02.10.3 and newer. Refer to [our FAQ](http://yt-dl.org/update) if you have trouble updating youtube-dl.

**Comment by operabeats** [comment]
it works now
As always you are incredibly fast! Thank you..!
